src/ghdcbot/config/loader.py

```
@@ -0,0 +1,58 @@
+
+from ghdcbot.config.models import BotConfig
+from ghdcbot.core.errors import ConfigError
+
+_ACTIVE_CONFIG: BotConfig | None = None
+_ENV_PATTERN = re.compile(r"\$\{([A-Z0-9_]+)\}")
+
+
+def load_config(path: str) -> BotConfig:
+    config_path = Path(path)
+    if not config_path.exists():
+        raise ConfigError(f"Config file does not exist: {path}")
```

Ruff (0.14.13)

Avoid specifying long messages outside the exception class

(TRY003)